src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
 class Herbivore extends Animal {
-  // write your code here
-}
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
 
+  hide() {
+    this.hidden = true;
+  }
+}
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim.hasOwnProperty('hidden') && !victim.hidden) {
+      victim.health -= 50;
+    }
+
+    if (victim.health === 0) {
+      Animal.alive.splice(Animal.alive.indexOf(victim), 1);
+    }
```

Comment:
The check for `victim.health === 0` is not sufficient to remove dead animals. The health could be less than 0 after a bite. Also, using `indexOf` and `splice` inside a loop can lead to performance issues. It's better to use `Array.prototype.filter` to remove dead animals from the `Animal.alive` array.